include/nuttx/serial/serial.h

```diff
@@ -283,10 +283,10 @@ struct uart_dev_s
   pid_t                pid;          /* Thread PID to receive signals (-1 if none) */
 #endif
 
-#ifdef CONFIG_SERIAL_TERMIOS
   /* Terminal control flags */
 
   tcflag_t             tc_iflag;     /* Input modes */
+#ifdef CONFIG_SERIAL_TERMIOS
```

Comment:
revert the change, let's check isconsole to do the input conversion if CONFIG_SERIAL_TERMIOS isn't enabled.

Reply:
That means if user want to disable the input conversion, them must enable TERMIOS to do that?

Reply:
isconsole control the default behaviour:

1. If isconsole equals true, the new line conversion for both direction and the check of Ctrl+C check will happen.
2. If isconsole equals false, no interception happens in serial driver level.

CONFIG_SERIAL_TERMIOS enable the user change the default behaviour at the runtime.

Reply:
Get it